### PR TITLE
Implement isEqual method

### PR DESCRIPTION
**Pull request by Augustyniak**
Description: Providing the "correct" implementation of `isEqual` method for `Element` object. This should help us for cases when we want have unit tests that want to verify whether a metrics with right `Element`'s is emitted or not.
Risk Level: Small
Testing: Unit Tests
Docs Changes: None
Release Notes: None

### CODE REVIEW

File: library/swift/test/BUILD [hunk 1]
```diff
@@ -77,3 +77,10 @@ envoy_mobile_swift_test(
         "CounterTests.swift",
     ],
 )
+
+envoy_mobile_swift_test(
+    name = "element_tests",
+    srcs = [
+        "ElementTests.swift",
+    ],
+)
```
Comment (rebello95) [inline library/swift/test/BUILD:86]: Trailing space 😃

File: library/swift/src/stats/Element.swift [hunk 2]
```diff
@@ -14,4 +14,12 @@ public final class Element: NSObject, ExpressibleByStringLiteral {
     }
     self.value = value
   }
+
+  public override func isEqual(_ object: Any?) -> Bool {
+    if let object = object as? Element {
+      return self.value == object.value
+    } else {
+      return false
+    }
```
Comment (rebello95) [inline library/swift/src/stats/Element.swift:23]:
```suggestion
    return (object as? Element)?.value == self.value
```

File: library/swift/test/ElementTests.swift [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+final class ElementTests: XCTestCase {
+    func testElementNotEqualToObjectOfUnrelatedType() {
+        let element = Element("foo")
+        XCTAssertFalse(element.isEqual("bar"))
+    }
+
+    func testElementsEquality() {
+      XCTAssertEqual(Element("foo"), Element("foo"))
+      XCTAssertNotEqual(Element("foo"), Element("bar"))
```
Comment (rebello95) [inline library/swift/test/ElementTests.swift:12]: Should we explicitly do an `isEqual` check here too?

File: library/swift/test/ElementTests.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import Envoy
+import XCTest
+
+final class ElementTests: XCTestCase {
+    func testElementNotEqualToObjectOfUnrelatedType() {
+        let element = Element("foo")
+        XCTAssertFalse(element.isEqual("bar"))
+    }
+
+    func testElementsEquality() {
```
Comment (rebello95) [inline library/swift/test/ElementTests.swift:10]:
```suggestion
    func testElementsAreEqualIfStringValuesAreEqual() {
```

File: library/swift/test/ElementTests.swift (hunk 3 above)
Comment (Augustyniak) [reply library/swift/test/ElementTests.swift:12]: maybe lets add 2 additional assertions - they would use `isEqual` method. WDYT?

File: library/swift/test/BUILD (hunk 1 above)
Comment (Augustyniak) [reply library/swift/test/BUILD:86]: it wasn't here previously so I though that it was not needed 🤔

Comment (rebello95) [reply library/swift/test/BUILD:86]: Screen Shot 2020-09-18 at 3 42 46 PM

I was referring to this ^ (not showing up on the main GitHub screen for some reason, only when you view files)

File: library/swift/test/ElementTests.swift (hunk 3 above)
Comment (rebello95) [reply library/swift/test/ElementTests.swift:12]: sure

**Review by rebello95 — DISMISSED** [review]

**Review by rebello95 — DISMISSED** [review]
Thanks!

**Review by rebello95 — APPROVED** [review]

**Review by buildbreaker — APPROVED** [review]

**Pull request merged**